src/platform/src/Bridge/Azure/OpenAI/EmbeddingsModelClient.php

```diff
@@ -34,11 +33,11 @@ public function __construct(
         #[\SensitiveParameter] private string $apiKey,
     ) {
         $this->httpClient = $httpClient instanceof EventSourceHttpClient ? $httpClient : new EventSourceHttpClient($httpClient);
-        Assert::notStartsWith($baseUrl, 'http://', 'The base URL must not contain the protocol.');
-        Assert::notStartsWith($baseUrl, 'https://', 'The base URL must not contain the protocol.');
-        Assert::stringNotEmpty($deployment, 'The deployment must not be empty.');
-        Assert::stringNotEmpty($apiVersion, 'The API version must not be empty.');
-        Assert::stringNotEmpty($apiKey, 'The API key must not be empty.');
+        !str_starts_with($this->baseUrl, 'http://') || throw new \InvalidArgumentException('The base URL must not contain the protocol.');
+        !str_starts_with($this->baseUrl, 'https://') || throw new \InvalidArgumentException('The base URL must not contain the protocol.');
+        '' !== $deployment || throw new \InvalidArgumentException('The deployment must not be empty.');
+        '' !== $apiVersion || throw new \InvalidArgumentException('The API version must not be empty.');
+        '' !== $apiKey || throw new \InvalidArgumentException('The API key must not be empty.');
```

Comment:
Let's us the component specific exception here please: `Symfony\AI\Platform\Exception\InvalidArgumentException`

Reply:
Indeed I had missed this class.
I have updated it wherever necessary.